h/services/user_delete.py

```diff
@@ -1,70 +1,315 @@
-import sqlalchemy as sa
+import logging
 
-from h.models import Annotation, Group, Token, User, UserDeletion
-from h.services.annotation_delete import AnnotationDeleteService
+from sqlalchemy import and_, delete, func, select, update
+
+from h.models import (
+    Annotation,
+    AuthTicket,
+    FeatureCohortUser,
+    Flag,
+    Group,
+    GroupMembership,
+    Job,
+    Token,
+    User,
+    UserDeletion,
+)
+
+log = logging.getLogger(__name__)
 
 
 class UserDeleteService:
-    def __init__(
-        self,
-        db_session: sa.orm.Session,
-        annotation_delete_service: AnnotationDeleteService,
-    ):
-        self._db = db_session
-        self._annotation_delete_service = annotation_delete_service
+    job_name = "purge_user"
```

Comment:
Hmm. I think this `JobName` enum is in the wrong place.

Having it on `JobQueueService` is a pain because `JobQueueService` is mocked in `UserDeleteService`'s tests, so `JobQueueService.JobName.PURGE_USER` is a mock not a string, which causes SQLAlchemy to crash when `UserDeleteService` tries to write a mock to the DB.

It can be fixed by moving `JobName` onto `models.Job` instead, which I think is where it belongs anyway: `JobQueueService` may not be the only code reading and writing the job table: https://github.com/hypothesis/h/pull/8700/commits/d8fcf1e695183e322e011cb023534a6f80e23cc0

I am slightly dubious about the value of these enums. The existing `JobName.SYNC_ANNOTATION` and `ANNOTATION_SLIM` aren't actually used anywhere! All the code just duplicates the strings.